ocaml/xapi/xapi_pvs_proxy.ml

```diff
@@ -22,8 +22,24 @@ let not_implemented x =
 let proxy_port_name bridge =
 	"pvs-" ^ bridge
 
+let get_accessible_cache_sr ~__context proxy =
+	let farm = Db.PVS_proxy.get_farm ~__context ~self:proxy in
+	let localhost = Helpers.get_localhost ~__context in
+	try
+		Some (List.find
+			(fun sr ->
+				Helpers.host_has_pbd_for_sr ~__context ~host:localhost ~sr)
+			(Db.PVS_farm.get_cache_storage ~__context ~self:farm))
+	with Not_found -> None
+
+let maybe_create_cache_vdi ~__context proxy =
+	match get_accessible_cache_sr ~__context proxy with
+	| Some sr -> ignore (Xapi_pvs_cache_vdi.find_or_create ~__context ~sr)
+	| None -> ()
```

Comment:
I'm not sure what the intended behaviour is - it could also be that there's no SR in the farm at all yet. Should that block VM startup, or should the VM be allowed to start with no cache?